src/config.cpp

```diff
@@ -196,6 +221,28 @@ struct ConfigParser<ConfigMissingProbes> {
   }
 };
 
+template <>
+struct ConfigParser<CompatibleBPFLicense> {
+  Result<OK> parse([[maybe_unused]] const std::string &key,
+                   CompatibleBPFLicense *target,
+                   const std::string &original)
+  {
+    auto found = BPF_LICENSE_STR.find(original);
+    if (found == BPF_LICENSE_STR.end()) {
+      return make_error<LicenseError>(original);
+    }
+
+    *target = found->second;
+    return OK();
+  }
+  Result<OK> parse(const std::string &key,
+                   CompatibleBPFLicense *target,
+                   uint64_t v)
+  {
+    return parse(key, target, std::to_string(v));
```

Comment:
Nit: this is fine, but I'd say this is unlikely to work out :)

Did you just want to return a ParseError here? E.g.

```
return make_error<ParseError>(key, "This does not appear to be a valid license: " + std::to_string(v));
```